src/platform_impl/android/mod.rs

```diff
@@ -502,15 +502,16 @@ impl<T: 'static> EventLoop<T> {
                         let pointers: Box<
                             dyn Iterator<Item = android_activity::input::Pointer<'_>>,
                         > = match phase {
-                            event::TouchPhase::Started
-                            | event::TouchPhase::Ended => {
+                            event::TouchPhase::Started => {
                                 Box::new(
                                     std::iter::once(motion_event.pointer_at_index(
-                                        motion_event.pointer_index(),
+                                        // In case it could be multi-touch gestures this requires the index of the last pointer.
+                                        motion_event.pointer_count() - 1_usize,
```

Comment:
It's not currently clear to me just from the comment why we should be assuming the last pointer is the correct one to reference here. Is it perhaps possible to quote some Android docs that clarify a bit why the index of the last pointer is appropriate here?

As far as I understand it from [these docs](https://developer.android.com/reference/android/view/MotionEvent#overview) the motion event's action code for `Up/PointerUp/Down/PointerDown` events should contain a pointer index (the action and pointer index are combined in a single number initially which need to be separated by masking and shifting) and that's what `.pointer_index()` does (it extracts the pointer index that's encoded in the action code).

_(`Up/Down` events should implicitly relate to pointer index zero)_

The `PointerDown` events seem pretty _likely_ to correspond to the last pointer but the Android docs don't seem to promise that as far as I can tell. The e.g. say "The order in which individual pointers appear within a motion event is undefined".

Reply:
it could be the fact that the order of iterator PointersIter is reversed.

there is also this doc about it :
https://developer.android.com/develop/ui/views/touch-and-input/gestures/multi
and/or  unmask  event like:
https://github.com/rust-mobile/android-activity/compare/main...haricot:android-activity:stylus_input

Reply:
Finally I do not think that the order is reversed, this is not a problem with a single touch since there is only one element. but in multi touch it seems to need a particular index.

I wonder if this is necessary to have access to the index  (or we already has with motion_event.action() &  motion_event.pointer_index()) :


[ACTION_POINTER_DOWN](https://developer.android.com/reference/androidx/core/view/MotionEventCompat#ACTION_POINTER_DOWN)—For extra pointers that enter the screen beyond the first. The index of the pointer that just went down can be obtained by using [getActionIndex()](https://developer.android.com/reference/androidx/core/view/MotionEventCompat#getActionIndex(android.view.MotionEvent)).

[ACTION_POINTER_UP](https://developer.android.com/reference/androidx/core/view/MotionEventCompat#ACTION_POINTER_UP)—Sent when a non-primary pointer goes up. The index of the pointer that just went up can be obtained by using [getActionIndex()](https://developer.android.com/reference/androidx/core/view/MotionEventCompat#getActionIndex(android.view.MotionEvent)).

` android-activity/game-activity-csrc/game-activity/native_app_glue/android_native_app_glue.c `


```
#define ACTION_POINTER_DOWN 0x00000005
#define ACTION_POINTER_UP 0x00000006

static bool default_motion_filter(const GameActivityMotionEvent* event) {
    // Ignore any non-touch events.
    // return event->source == SOURCE_TOUCHSCREEN;
    return (event->source == SOURCE_TOUCHSCREEN ||
event-> action == ACTION_POINTER_DOWN
event-> action ==  ACTION_POINTER_UP
);
}
```